src/main.rs

```diff
@@ -1,3 +1,120 @@
+use {
+    rand::Rng,
+    std::{cmp, io},
+};
+
+pub fn print_car_name_format() {
+    println!("경주할 자동차 이름을 입력하세요.(이름은 쉼표(,) 기준으로 구분)");
+}
+
+pub fn get_car_names_from_input() -> Vec<String> {
+    let mut cars: Vec<String> = Vec::new();
+
+    let mut line = String::new();
+    io::stdin()
+        .read_line(&mut line)
+        .expect("잘못된 입력입니다.");
+
+    let splited: Vec<&str> = line.trim().split(',').collect();
+    for i in 0..splited.len() as i64 {
+        cars.push(splited[i as usize].to_owned());
+    }
+    cars
+}
+
+pub fn print_total_round_format() {
+    println!("전체 라운드 수는 몇회인가요?");
+}
+
+pub fn get_total_round_from_input() -> i64 {
+    loop {
+        let mut line = String::new();
+        io::stdin()
+            .read_line(&mut line)
+            .expect("잘못된 입력입니다.");
+        let total_round = match line.trim().parse::<i64>() {
+            Ok(total_round) => total_round,
+            Err(_) => {
+                eprintln!("[Error] 시도 횟수는 숫자여야 한다.");
+                continue;
+            }
+        };
+        return total_round;
+    }
+}
+
+pub fn print_result_per_round(car_names: Vec<String>, dist_info: Vec<String>) {
+    let cars_num: i64 = car_names.len() as i64;
+    for car_num in 0..cars_num {
+        println!(
+            "{} : {}",
+            car_names[car_num as usize], dist_info[car_num as usize]
+        );
+    }
+    println!();
+}
+
+pub fn run_round(car_names: Vec<String>, dist_info: &mut [String]) {
+    let mut rng = rand::thread_rng();
```

Comment:
Random variables make it difficult to write tests because they are code over which we have no control. Let's pull this out and turn it into testable code.